### DISCUSSION TITLE
Ideas for owncast.

### DISCUSSION THREAD

**Discussion by sethschn** [owncast/owncast#3408] [Ideas]
I've been using Owncast and I love it. Some things I would like to see are banned people not being able to watch the video and Discord authentication built in.

**Comment by gabek** [comment]
Hi there, thanks for the suggestions.

Banning people from viewing is actually more complicated than it seems. It's straightforward on a basic Owncast setup, where Owncast is the serving the video directly. But in the case somebody is using object storage or puts a CDN in front of their server then the requests never even get to the Owncast server, there's no way to block them.

I'm not exactly sure what you mean by Discord authentication, but there are no plans to put any kind of proprietary user integration into Owncast.

**Comment by sethschn** [comment]
You know how when you authenticate. Instead of that it should be discord. Discord provides oauth2 authentication.

**Comment by gabek** [comment]
I understand you like Discord, but to suggest functionality be removed from owncast and replaced with hard-coding a dependency to Big Tech is a bit tasteless.

**Reply by sethschn** [reply]
To not support oauth2 is a bit tasteless don't you think? I get it you hate discord and its your project, but you don't have to be rude about it.
